src/annotator/index.js

```diff
@@ -33,34 +33,28 @@ const appLinkEl = /** @type {Element} */ (
   )
 );
 
-const config = configFrom(window);
-
 function init() {
+  const annotatorConfig = getConfig('annotator');
```

Comment:
The original solution I proposed was a class instance, but  I got feedback that steered this in another direction
The major push back was making the API seem like the config would change the lifetime of the instance and perhaps not be deterministic.  More discussion here.

https://github.com/hypothesis/client/pull/3232#pullrequestreview-626217791

Reply:
I think Lyza's feedback in https://github.com/hypothesis/client/pull/3232#pullrequestreview-626217791 was quite reasonable. However I don't interpret it as a hard request not to use a class but instead a request to 1) justify its value and 2) clarify when parsing occurs. I think I would respond to point (2) by just adjusting the API design/docs/implementation to make it clear when parsing happens and regarding (1) I think it's clear from this PR that it results in doing parsing / validation once rather than thrice.

I think making any further changes in response to this comment can be the subject of a later PR though.

```js
class Config {
  constructor(allSettings) { ... }
  
  configFor(context) {
    // Return subset of `allSettings` for current context
  }
}

// `parseConfig` name makes it clear that configuration extraction/parsing happens
// when this function is called.
export function parseConfig(window_) {
  const allSettings = readAllTheSettings();
  return new Config(allSettings);
}
```